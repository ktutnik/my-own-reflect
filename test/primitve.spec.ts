import reflect from "../src"

describe("Primitive Class", () => {
    it("Should inspect Boolean", () => {
        expect(reflect(Boolean)).toMatchSnapshot()
    })
    it("Should inspect String", () => {
        expect(reflect(String)).toMatchSnapshot()
    })
    it("Should inspect Number", () => {
        expect(reflect(Number)).toMatchSnapshot()
    })
    it("Should inspect Date", () => {
        expect(reflect(Date)).toMatchSnapshot()
    })
    it("Should inspect Array", () => {
        expect(reflect(Array)).toMatchSnapshot()
    })
    it("Should inspect Object", () => {
        expect(reflect(Object)).toMatchSnapshot()
    })
    it("Should inspect Promise", () => {
        expect(reflect(Promise)).toMatchSnapshot()
    })
})