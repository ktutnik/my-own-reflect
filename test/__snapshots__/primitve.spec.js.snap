// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primitive Class Should inspect Array 1`] = `
Object {
  "ctor": Object {
    "kind": "Constructor",
    "name": "constructor",
    "parameters": Array [],
  },
  "decorators": Array [],
  "kind": "Class",
  "methods": Array [
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "concat",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "find",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "findIndex",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "pop",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "push",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "shift",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "unshift",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "slice",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "splice",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "includes",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "indexOf",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "keys",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "entries",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "forEach",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "filter",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "map",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "every",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "some",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "reduce",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "reduceRight",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toString",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toLocaleString",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "join",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "reverse",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "sort",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "lastIndexOf",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "copyWithin",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "fill",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "values",
      "owner": Array [
        Array,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
  ],
  "name": "Array",
  "owner": Array [
    Object,
    Array,
  ],
  "properties": Array [],
  "super": Object,
  "type": Array,
  "typeClassification": "Class",
}
`;

exports[`Primitive Class Should inspect Boolean 1`] = `
Object {
  "ctor": Object {
    "kind": "Constructor",
    "name": "constructor",
    "parameters": Array [],
  },
  "decorators": Array [],
  "kind": "Class",
  "methods": Array [
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toString",
      "owner": Array [
        Boolean,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "valueOf",
      "owner": Array [
        Boolean,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
  ],
  "name": "Boolean",
  "owner": Array [
    Object,
    Boolean,
  ],
  "properties": Array [],
  "super": Object,
  "type": Boolean,
  "typeClassification": "Class",
}
`;

exports[`Primitive Class Should inspect Date 1`] = `
Object {
  "ctor": Object {
    "kind": "Constructor",
    "name": "constructor",
    "parameters": Array [],
  },
  "decorators": Array [],
  "kind": "Class",
  "methods": Array [
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toString",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toDateString",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toTimeString",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toISOString",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toUTCString",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toGMTString",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getDate",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setDate",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getDay",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getFullYear",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setFullYear",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getHours",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setHours",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getMilliseconds",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setMilliseconds",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getMinutes",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setMinutes",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getMonth",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setMonth",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getSeconds",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setSeconds",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getTime",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setTime",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getTimezoneOffset",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getUTCDate",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setUTCDate",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getUTCDay",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getUTCFullYear",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setUTCFullYear",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getUTCHours",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setUTCHours",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getUTCMilliseconds",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setUTCMilliseconds",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getUTCMinutes",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setUTCMinutes",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getUTCMonth",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setUTCMonth",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getUTCSeconds",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setUTCSeconds",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "valueOf",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "getYear",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "setYear",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toJSON",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toLocaleString",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toLocaleDateString",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toLocaleTimeString",
      "owner": Array [
        Date,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
  ],
  "name": "Date",
  "owner": Array [
    Object,
    Date,
  ],
  "properties": Array [],
  "super": Object,
  "type": Date,
  "typeClassification": "Class",
}
`;

exports[`Primitive Class Should inspect Number 1`] = `
Object {
  "ctor": Object {
    "kind": "Constructor",
    "name": "constructor",
    "parameters": Array [],
  },
  "decorators": Array [],
  "kind": "Class",
  "methods": Array [
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toExponential",
      "owner": Array [
        Number,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toFixed",
      "owner": Array [
        Number,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toPrecision",
      "owner": Array [
        Number,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toString",
      "owner": Array [
        Number,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "valueOf",
      "owner": Array [
        Number,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toLocaleString",
      "owner": Array [
        Number,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
  ],
  "name": "Number",
  "owner": Array [
    Object,
    Number,
  ],
  "properties": Array [],
  "super": Object,
  "type": Number,
  "typeClassification": "Class",
}
`;

exports[`Primitive Class Should inspect Object 1`] = `
Object {
  "ctor": Object {
    "kind": "Constructor",
    "name": "constructor",
    "parameters": Array [],
  },
  "decorators": Array [],
  "kind": "Class",
  "methods": Array [
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "hasOwnProperty",
      "owner": Array [
        Object,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "isPrototypeOf",
      "owner": Array [
        Object,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "propertyIsEnumerable",
      "owner": Array [
        Object,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toString",
      "owner": Array [
        Object,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "valueOf",
      "owner": Array [
        Object,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toLocaleString",
      "owner": Array [
        Object,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
  ],
  "name": "Object",
  "owner": Array [
    Object,
    Object,
  ],
  "properties": Array [],
  "super": Object,
  "type": Object,
  "typeClassification": "Class",
}
`;

exports[`Primitive Class Should inspect Promise 1`] = `
Object {
  "ctor": Object {
    "kind": "Constructor",
    "name": "constructor",
    "parameters": Array [],
  },
  "decorators": Array [],
  "kind": "Class",
  "methods": Array [
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "then",
      "owner": Array [
        Promise,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "catch",
      "owner": Array [
        Promise,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "finally",
      "owner": Array [
        Promise,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
  ],
  "name": "Promise",
  "owner": Array [
    Object,
    Promise,
  ],
  "properties": Array [],
  "super": Object,
  "type": Promise,
  "typeClassification": "Class",
}
`;

exports[`Primitive Class Should inspect String 1`] = `
Object {
  "ctor": Object {
    "kind": "Constructor",
    "name": "constructor",
    "parameters": Array [],
  },
  "decorators": Array [],
  "kind": "Class",
  "methods": Array [
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "anchor",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "big",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "blink",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "bold",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "charAt",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "charCodeAt",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "codePointAt",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "concat",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "endsWith",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "fontcolor",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "fontsize",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "fixed",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "includes",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "indexOf",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "italics",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "lastIndexOf",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "link",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "localeCompare",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "match",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "normalize",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "padEnd",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "padStart",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "repeat",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "replace",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "search",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "slice",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "small",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "split",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "strike",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "sub",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "substr",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "substring",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "sup",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "startsWith",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toString",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "trim",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "trimLeft",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "trimRight",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toLowerCase",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toUpperCase",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "valueOf",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toLocaleLowerCase",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "toLocaleUpperCase",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "trimStart",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
    Object {
      "decorators": Array [],
      "kind": "Method",
      "name": "trimEnd",
      "owner": Array [
        String,
      ],
      "parameters": Array [],
      "returnType": undefined,
      "typeClassification": undefined,
    },
  ],
  "name": "String",
  "owner": Array [
    Object,
    String,
  ],
  "properties": Array [],
  "super": Object,
  "type": String,
  "typeClassification": "Class",
}
`;
